dataprofiler/data_readers/filepath_or_buffer.py

```diff
@@ -43,7 +47,14 @@ def __enter__(self):
         return self._filepath_or_buffer
 
     def __exit__(self, exc_type, exc_value, exc_traceback):
-        if isinstance(self._filepath_or_buffer, (StringIO, BytesIO)):
+        if isinstance(self._filepath_or_buffer, (StringIO, BytesIO, DetachingTextIOWrapper)):
             self._filepath_or_buffer.seek(0)
         else:
             self._filepath_or_buffer.close()
+
+class DetachingTextIOWrapper(TextIOWrapper):
```

Comment:
Can you give this class docstrings?

Reply:
When you do could you mention the use case too, I don't immediately understand the application by just reading the code.